opentracing-api/src/main/java/io/opentracing/SpanContext.java

```diff
@@ -26,6 +26,28 @@
  * @see Span#getBaggageItem(String)
  */
 public interface SpanContext {
+    /**
+     * Return the ID of the trace.
+     *
+     * Globally unique. Every span in a trace shares this ID.
+     *
+     * The empty string is a valid return value, while null is not.
```

Comment:
Do we need to clarify what an empty string represents? i.e. no active trace instance.
Also do we need other text that indicates the return value of toTraceId and toSpanId are either both empty strings (no active trace) or both have ids? For example, I don't think it would be valid to have an empty trace id but a value returned from span id?

Reply:
Oh, that's a good point. IMHO it would make sense to mention that the empty string *ought* to represent missing Trace Identifiers info (and, if this gets approved, make also the note about Span ID & Trace ID working together).

Sounds like a good plan?

cc @yurishkuro @tedsuo

Reply:
iirc the original idea was to allow empty string to tracers that are unable to provide a single trace id (e.g. "do not support them", or more likely "this span has many trace IDs").

Reply:
I would recommend we keep the rules simple, and just say that either method may return an empty string, meaning that the method is not supported for that span. 

I doubt implementations will provide one but not the other, but there's no need to make it a rule.